### PR TITLE
fix: cext-3092: add log:list command

### PR DESCRIPTION
**Pull request by dani97**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/api-mesh/log-list.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	async run() {
+		await initRequestId();
+
+		logger.info(`RequestId: ${global.requestId}`);
+
+		const { flags } = await this.parse(ListLogsCommand);
+
+		const ignoreCache = await flags.ignoreCache;
+		const json = await flags.json;
+		const fileName = await flags.filename;
```
Comment (amolina-adobe) [inline src/commands/api-mesh/log-list.js:31]: Don't need the `await` keywords here.
```suggestion
		const { ignoreCache, json, filename: fileName } = flags;
```

File: src/commands/api-mesh/log-list.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+							...flags,
+						},
+					);
+				} else {
+					this.log(
+						'No recent logs found. Alternatively, you can use the following command to get all logs for a 30 minute time period: \naio api-mesh log-get-bulk --startTime YYYY-MM-DDTHH:MM:SSZ --endTime YYYY-MM-DDTHH:MM:SSZ --filename mesh_logs.csv',
+					);
+				}
+			} catch (error) {
+				this.error(`Failed to fetch logs, RequestId: ${global.requestId}`);
```
Comment (amolina-adobe) [inline src/commands/api-mesh/log-list.js:98]:
```suggestion
				this.error(`Failed to list recent logs, RequestId: ${global.requestId}`);
```

File: src/utils.js [hunk 3]
```diff
@@ -67,6 +67,10 @@ const selectFlag = Flags.boolean({
 	default: false,
 });
 
+const fileNameFlag = Flags.string({
+	description: 'Name of Csv file for the recent logs',
```
Comment (amolina-adobe) [inline src/utils.js:71]:
```suggestion
	description: 'Name of CSV file to export the recent logs to',
```

**Review by amolina-adobe — APPROVED** [review]
Looks great 👍  Comments are optional.

File: src/commands/api-mesh/log-list.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		const json = await flags.json;
+		const fileName = await flags.filename;
+
+		if (fileName) {
+			if (path.extname(fileName).toLowerCase() !== '.csv') {
+				this.error('Invalid file type. Provide a filename with a .csv extension.');
+			}
+			const file = path.resolve(process.cwd(), fileName);
+			if (existsSync(file)) {
+				this.error(`File ${fileName} already exists. Please provide a new file name.`);
```
Comment (dthampy) [inline src/commands/api-mesh/log-list.js:39]: @dani97, please remove all occurrences of "Please" from the messages.
@jhadobe, aside from the use of "Please," do you agree with the wording of the messages?

File: src/commands/api-mesh/log-list.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+const { appendFileSync, existsSync } = require('fs');
+const { ux } = require('@oclif/core/lib/cli-ux');
+const path = require('path');
+
+require('dotenv').config();
+class ListLogsCommand extends Command {
+	//static args = [{ name: 'file' }];
+	static flags = {
+		ignoreCache: ignoreCacheFlag,
+		json: jsonFlag,
```
Comment (dthampy) [inline src/commands/api-mesh/log-list.js:16]: @dani97 what does this jsonFlag do? write it in json format in the csv file?

File: src/commands/api-mesh/log-list.js (hunk 4 above)
Comment (jhadobe) [reply src/commands/api-mesh/log-list.js:39]: @dthampy yes, looks good after the "Please" is removed.

File: src/commands/api-mesh/log-list.js (hunk 5 above)
Comment (dani97) [reply src/commands/api-mesh/log-list.js:16]: Going to remove this, I took reference from other commands not needed here

Comment (dani97) [reply src/commands/api-mesh/log-list.js:16]: they use it for more verbose output in json. here we wont need it

**Pull request merged**